sklearn/model_selection/tests/test_validation.py

```diff
@@ -754,6 +755,23 @@ def test_permutation_test_score_allow_nans():
     permutation_test_score(p, X, y)
 
 
+def test_permutation_test_score_fit_params():
+    est = SVC(random_state=0)
+    n_samples = 50
+    X, y = make_blobs(n_samples=n_samples, n_features=2, centers=10,
+                      random_state=0)
+    weights = [i for i in range(n_samples)]
+    score_without_fit_params = permutation_test_score(est, X, y)
+    mean_without_fit_params = score_without_fit_params[1].mean()
+    # checking for sample weight with list and numpy array
+    for method in [list, np.array]:
+        w = method(weights)
```

Comment:
For the scope of this PR, I do not think we need to check for both list and ndarray.

Reply:
What do you suggest? Simplify the tests by only testing with arrays for the sample weights? I agree that in any case it's up to the underlying estimators to handle the specific type of the fit params so it should not matter.

Alternatively we could make a more explicit test about the routing of the fit params by introducing a private test helper class such as:

```
class _FitParamsRecordingClassifier(DummyClassifier):
    def fit(self, X, y, **fit_params):
        super().fit(X, y, sample_weight=fit_params.get("sample_weight"))
        self.fit_params_ = fit_params
```

Reply:
I would prefer something explicit. Our `CheckingEstimator` has the capability to check for this:

```py
CheckingClassifier(expected_fit_params=['sample_weight'])
```

You can see it usaged in tests here:

https://github.com/scikit-learn/scikit-learn/blob/0864c5896804bc1066c02bcb1443c962cabdc420/sklearn/model_selection/tests/test_search.py#L225-L243

Reply:
I’ll try to fix these over the weekend

Reply:
fixed these using the suggested classifier. thanks @thomasjpfan  and @ogrisel for your comments.